src/TrigScint/QIEAnalyzer.cxx

```diff
@@ -1,184 +1,215 @@
-    inputPassName_  = parameters.getParameter< std::string >("inputPassName");
-    peds_  = parameters.getParameter< std::vector<double> >("pedestals");
-    gain_  = parameters.getParameter< std::vector<double> >("gain");
-    startSample_  = parameters.getParameter< int >("startSample");
-
-    std::cout << " [ QIEAnalyzer ] In configure(), got parameters " 
-	      << "\n\t inputCollection = " << inputCol_
-	      << "\n\t inputPassName = " << inputPassName_
-	      << "\n\t startSample = " << startSample_
-	      << "\n\t pedestals[0] = " << peds_[0]
-	      << "\n\t gain[0] = " << gain_[0]
-	      << "\t." << std::endl;
-
-    return;
-  }
 
-  void QIEAnalyzer::analyze(const framework::Event &event) {
-
-    const auto channels{event.getCollection<trigscint::EventReadout >(inputCol_, inputPassName_)};
-
-	int evNb = event.getEventNumber();
-	//	while (evNb < 0 ) {
-	//  ldmx_log(debug) << "event number = " << evNb << " < 0; incrementing event number ";
-	//  evNb++;
-	//}
-	int nChan = channels.size();
-	ldmx_log(debug) << "in event " << evNb << "; nChannels = " << nChan;
-	
-	for (auto chan : channels) {
-	  std::vector<float> q = chan.getQ();
-	  std::vector<float> qErr = chan.getQError();
-	  std::vector<int> tdc = chan.getTDC();
-	  int nTimeSamp = q.size();
-	  int bar = chan.getChanID();
-	  float qTot = 0;
-	  float qPedSubtractedAvg = 0;
-	  int firstT = startSample_-1;
-	  int nSampAbove = 0;
-	  int nSampAboveEventPed = 0;
-	  float subtrPE = 0;
-	  float subtrQ = 0;
-	  float ped = chan.getPedestal() ;
-	  for (int iT = 0; iT < q.size() ; iT++) {
-		ldmx_log(debug) << "in event " << evNb << "; channel " << bar << ", got charge[" << iT << "] = " << q.at(iT);
-		if ( evNb < nEv && bar < nChannels ) { //stick within the predefined histogram array
-		  //		  hOut[evNb][bar]->Fill(iT+startSample_, q.at(iT));
-		  hOut[evNb][bar]->SetBinContent(iT+startSample_, q.at(iT));
-		  hOut[evNb][bar]->SetBinError(iT+startSample_, fabs(qErr.at(iT)) );
-		  if ( tdc.at(iT) < 63 ) {
-			ldmx_log(info) << "Found fired TDC = " << tdc.at(iT) << " at time sample " << iT << " in channel " << bar << " and event " << evNb ;
-			hOut[evNb][bar]->SetLineColor(kRed+1); //for some reason, the style settings are washed out later...
-			hOut[evNb][bar]->SetMarkerColor(hOut[evNb][bar]->GetLineColor());
-			hOut[evNb][bar]->SetMarkerSize(0.2);
-
-			if (iT+startSample_  > 0)
-			  hTDCfireChanvsEvent->Fill( bar, evNb, iT+startSample_);
-			else 
-			  hTDCfireChanvsEvent->Fill( bar, evNb);
-		  }
-		}//if within the number of events to plot individually 
-		if ( q.at(iT) > 2*fabs(peds_[ bar ]) ) 	{ //integrate all charge well above ped to convert to a PE count
-		  qTot+=q.at(iT);
-		  qPedSubtractedAvg+=q.at(iT)-chan.getPedestal(); //peds_[ bar ];
-		  nSampAbove++;
-		  ldmx_log(debug) << " above channel overall pedestal: " << q.at(iT) << " > " << 2*fabs(peds_[bar]) ;
-   
-		  if (firstT = startSample_-1) //keep track of first time sample above threshold
```

Comment:
This wasn't changed by me, but is this assignment on purpose?

Reply:
sorry, i don't get what you mean here

Reply:
The if-statement uses an = rather than an ==, is that on purpose? I didn't change anything here I just noticed it while going through things